### PR TITLE
chore: upgrade Prettier and do baseline formatting of all files

### PR DESCRIPTION
**Pull request by ryami333**
At this moment there are a bunch of files in this project which have not been formatted with Prettier, which would cause `npx prettier . --check` to fail, and would make subsequent pull-request diffs noisy, because they might contain formatting changes unrelated to the PR's subject-matter.

So in this pull-request I have:
1. Bumped Prettier to the latest version.
2. Updated the `prettier` NPM script to format _all_ supported files, not just JS, TS and Markdown files.
3. Ran `npx prettier . --write` to perform a baseline formatting of the entire project.

### CODE REVIEW

File: package.json
Comment (ryami333) [inline package.json:36]: In Prettier's very early days you had to explicitly specify which files to check/format like this, but since Prettier 2.0 you can just pass `.` to expand the current directory, and Prettier will check+format all _compatible_ files (and ignore any incompatible ones).

File: playground-nextjs/next.config.js
Comment (ryami333) [inline playground-nextjs/next.config.js:1]: All changes you see from this file and below are just formatting changes that were made programmatically by running `npx prettier . --write` (or `npm run prettier`).

**Comment by ryami333** [comment]
@alvarosabu could we (Diesdas) please have an update on this status of my PRs?

File: package-lock.json
Comment (alvarosabu) [inline package-lock.json:1]: Hi @ryami333 did you bump the prettier version with `--save`? It seems like it only updated the lock but not the package.json itself. 

I think the latest prettier is around `v3.3.x`

File: playground-nextjs/next.config.js
Comment (alvarosabu) [inline playground-nextjs/next.config.js:3]: Since we are fixing the formatting let's go further and remove the gigantic indentation on the files. I did already on a separate refactor PR but i will leave to your call @ryami333 if you want to add it to the PR.

Steps: 
 -Open root `package.json`
 - Go down to the property `prettier` and remove `useTabs`
 - Run `npm run prettier` again

The PR files will increase to around ~44 but since they are only identation changes its gonna be easier to check.

Screenshot 2024-07-08 at 10 45 38

**Review by alvarosabu — CHANGES_REQUESTED** [review]
Hi @ryami333 thank you so much for taking the time for submitting this PR. We are currently refactoring the setup of the repo in a WIP branch but if we can tackle small things like this to unblock you guys the better for both. 

I added a couple of things to the review, let me know if you have any questions.

File: playground-nextjs/next.config.js
Comment (ryami333) [reply playground-nextjs/next.config.js]: Done ✅

Comment (ryami333) [reply playground-nextjs/next.config.js]: I misinterpreted the request initially - I inadvertently removed the `trailingComma` rule as well. I've just re-instated that and pushed.

**Review by alvarosabu — APPROVED** [review]

**Comment by ryami333** [comment]
Yep cool – I'm not authorised to merge the PR, so it's over to you in any case 👍

**Comment by alvarosabu** [comment]
> Also, the tests can't be run locally because they depend on some environment variables that non-Storyblok staff wouldn't have access to, so it's not a particularly open-source-friendly setup.

I totally agree, my plan is to mock the requests on the unit test in the near future. Thanks for the valuable feedback 🙏🏻

**Pull request merged**
